### PR TITLE
Add link to voting

### PR DESCRIPTION
**Pull request by kozlovzxc-at-lido**
## Description

Add link to view voting details

## Demo
For localhost it is vote.lido.fi
Screenshot 2023-05-23 at 14 17 15

For localhost + voteId it is vote.lido.fi/vote/${voteId}
Screenshot 2023-05-23 at 14 18 45

For custom domain it's tld and the same rules
Screenshot 2023-05-23 at 14 19 58

## Testing notes

For testnet it should be vote.testnet.fi, for stage and prod same rules.

### CODE REVIEW

File: shared/ui/votingLink/votingLink.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+export const VotingLink: FC<PropsWithChildren<{ voteId?: string }>> = memo(
+  ({ voteId, children }) => {
+    const [href, setHref] = useState('https://vote.lido.fi');
+
+    useEffect(() => {
+      const url = new URL(window.location.href);
+      const tld =
+        url.hostname === 'localhost'
+          ? 'lido.fi'
+          : url.hostname.split('.').slice(-2).join('.');
```
Comment (Jabher) [inline shared/ui/votingLink/votingLink.tsx:13]: will this work OK with preview stands?

File: shared/ui/votingLink/votingLink.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        voteId == null ||
+        voteId === '' ||
+        Number.isNaN(Number(voteId)) ||
+        voteId !== Math.trunc(Number(voteId)).toString()
+          ? ''
+          : `/vote/${voteId}`;
+      setHref(`https://vote.${tld}${path}`);
+    }, [voteId]);
+
+    return <Link href={href}>{children}</Link>;
```
Comment (Jabher) [inline shared/ui/votingLink/votingLink.tsx:24]: maybe just render nothing if no location is present...?

File: shared/ui/votingLink/votingLink.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      const url = new URL(window.location.href);
+      const tld =
+        url.hostname === 'localhost'
+          ? 'lido.fi'
+          : url.hostname.split('.').slice(-2).join('.');
+      const path =
+        voteId == null ||
+        voteId === '' ||
+        Number.isNaN(Number(voteId)) ||
+        voteId !== Math.trunc(Number(voteId)).toString()
```
[the remark this answers is not in the document]
Comment (Jabher) [reply shared/ui/votingLink/votingLink.tsx:18]: also, is it matching with other validations? 
maybe, you can eject the check and re-use it so we will not have same logic twice?

File: shared/ui/votingLink/votingLink.tsx (hunk 1 above)
Comment (kozlovzxc-at-lido) [reply shared/ui/votingLink/votingLink.tsx:13]: We don't have preview stands on TRP now and probably there will be changes of how do they work in some foreseeable future 🤔 So I'm not sure if we need to add this case ahead of time

Comment (kozlovzxc-at-lido) [reply shared/ui/votingLink/votingLink.tsx:13]: Do you think it's reasonable only allow these changes for prod/testnet/stage and keep vote.lido.fi for all other runs?

File: shared/ui/votingLink/votingLink.tsx (hunk 2 above)
Comment (kozlovzxc-at-lido) [reply shared/ui/votingLink/votingLink.tsx:24]: Why? It's a bit better for SEO to have production value, and we don't really care if it's incorrect for testnet and stage

File: shared/ui/votingLink/votingLink.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (kozlovzxc-at-lido) [reply shared/ui/votingLink/votingLink.tsx]: Makes sense, used the same validation as on form

File: features/aragon/votingLink.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+export const VotingLink: FC<PropsWithChildren<{ voteId?: string }>> = memo(
+  ({ voteId, children }) => {
+    const [href, setHref] = useState('https://vote.lido.fi');
+
+    useEffect(() => {
+      const url = new URL(window.location.href);
+      const tld =
+        url.hostname === 'localhost'
+          ? 'lido.fi'
+          : url.hostname.split('.').slice(-2).join('.');
```
Comment (Jabher) [inline features/aragon/votingLink.tsx:14]: IDK why my same comment on this topic disappeared, but again... will it work with branch-preview?

Comment (kozlovzxc-at-lido) [reply features/aragon/votingLink.tsx:14]: image

It shows for me as outdated, maybe that's why

Comment (Jabher) [reply features/aragon/votingLink.tsx:14]: I would prefer to have some switch here with default fallback to lido.fi

File: shared/ui/votingLink/votingLink.tsx (hunk 2 above)
Comment (Jabher) [reply shared/ui/votingLink/votingLink.tsx]: sounds reasonable

**Review by Jabher — APPROVED** [review]

File: features/aragon/votingLink.tsx (hunk 4 above)
Comment (kozlovzxc-at-lido) [reply features/aragon/votingLink.tsx:14]: Makes sense, will add

**Pull request merged**